tests/core/go_path/pkg/lib/BUILD.bazel

```diff
@@ -10,13 +11,20 @@ go_library(
     ],
     embedsrcs = [
         "embedded_src.txt",
+        "renamed_embedded_src.txt",
         "template/index.html.tmpl",
     ],
     importpath = "example.com/repo/pkg/lib",
     visibility = ["//visibility:public"],
     deps = [":transitive_lib"],
 )
 
+copy_file(
```

Comment:
A simple test case you could add that would cover the case of a transition would be a `go_binary` built with a non-default `linkmode`.

Reply:
Added a test case that only passed with the f.root.path route you suggested. I'm not that familiar with the testing convention, so any tips to simplify the test setup would be appreciated.

Reply:
Thanks for making the changes, this already looks pretty good. 

Could we add one more test case where we turn this around and add a transitioned Go binary ("pie" or "cross" should work) as an `embedsrc` here?

Reply:
Thinking about it more, I don't think this test case would actually be relevant here.